libraries/core_libs/network/src/tarcap/threadpool/packets_blocking_mask.cpp

```diff
@@ -76,6 +76,28 @@ std::optional<taraxa::level_t> PacketsBlockingMask::getSmallestDagLevelBeingProc
   return {};
 }
 
+void PacketsBlockingMask::setDagBlockBeingProcessed(const PacketData& packet) {
+  // Signature is used as id for the dag block since it is cheaper to get signature than to calculate hash at this point
+  // and signature should be unique per dag block
+  sig_t sig = DagBlock::extract_signature_from_rlp(packet.rlp_);
+
+  // If blocking is working correctly it should not be possible that we already are processing the same block
+  const auto processing_dag_block = processing_dag_blocks_.find(sig);
+  assert(processing_dag_block == processing_dag_blocks_.end());
+
+  processing_dag_blocks_.emplace(std::move(sig), packet.id_);
+}
+
+void PacketsBlockingMask::unsetDagBlockBeingProcessed(const PacketData& packet) {
+  sig_t sig = DagBlock::extract_signature_from_rlp(packet.rlp_);
+
+  // There must be existing block inside processing_dag_blocks_
+  const auto processing_dag_block = processing_dag_blocks_.find(sig);
+  assert(processing_dag_block != processing_dag_blocks_.end());
+
+  processing_dag_blocks_.erase(sig);
```

Comment:
just a small speed up, I would use iterator `processing_dag_block` for erase as you do not need to search again

Reply:
Fixed